staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go

```diff
@@ -58,6 +58,13 @@ type Storage interface {
 	New() runtime.Object
 }
 
+// Scoper indicates what scope the resource is at. It must be specified.
+// It is usually provided automatically based on your strategy.
+type Scoper interface {
+	// NamespaceScoped returns true if the storage is namespaced
+	NamespaceScoped() bool
```

Comment:
Seems like this just needs to be part of `type Storage interface {` above?  It is required in all cases from what I can tell so why not make it a compile time check?